### PR TITLE
docs: Document limitation regarding subpaths

### PR DESCRIPTION
**Pull request by c3d**
Subpaths are not supported at the moment. Document that fact.

Fixes: #1728

Signed-off-by: Christophe de Dinechin <christophe@dinechin.org>

### CODE REVIEW

File: docs/Limitations.md [hunk 1]
```diff
@@ -224,7 +236,7 @@ Privileged support in Kata is essentially different from `runc` containers.
 Kata does support `docker run --privileged` command, but in this case full access
 to the guest VM is provided in addition to some host access.
 
-The container runs with elevated capabilities within the guest and is granted 
+The container runs with elevated capabilities within the guest and is granted
```
Comment (fidencio) [inline docs/Limitations.md:239]: This change is unrelated and should be dropped.
Otherwise, lgtm.  Thanks @c3d!

Comment (c3d) [reply docs/Limitations.md:239]: Emacs with auto-removal of trailing whitespaces. I'll submit it as a separate patch.

Comment (fidencio) [reply docs/Limitations.md:239]: I wouldn't even bother, or I'd advocate for a CI that checks for those (as the one from the OpenShift enhancement requests).
Otherwise it'll be a never-ending battle.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @c3d!

**Comment by Tim-Zhang** [comment]
@c3d There is a error in spelling check: `Word 'Subpaths': did you mean one of the following?: Sub paths, Sub-paths, Subpar`

File: docs/Limitations.md [hunk 2]
```diff
@@ -216,6 +218,16 @@ Equivalent functionality can be achieved with the newer docker networking comman
 See more documentation at
 [docs.docker.com](https://docs.docker.com/engine/userguide/networking/default_network/dockerlinks/).
 
+## Storage limitations
+
+### Subpaths
```
Comment (wainersm) [inline docs/Limitations.md:223]: I suggest to expand the title to something like "Kubernetes volumeMounts subPath".  More or less how it is called in https://kubernetes.io/docs/concepts/storage/volumes/#using-subpath

File: docs/Limitations.md [hunk 3]
```diff
@@ -216,6 +218,16 @@ Equivalent functionality can be achieved with the newer docker networking comman
...
+
+Kubernetes `subPath` is not supported by Kata at the moment.
```
Comment (wainersm) [inline docs/Limitations.md:225]: I think it was @jodh-intel who once said me to avoid calling the project simply "Kata" on user-facing text. Instead, "Kata Containers".

File: docs/Limitations.md (hunk 2 above)
Comment (c3d) [reply docs/Limitations.md]: Good idea. Done.

File: docs/Limitations.md (hunk 3 above)
Comment (c3d) [reply docs/Limitations.md]: Good point. Fixed.

**Comment by c3d** [comment]
> @c3d There is an error in spelling check: `Word 'Subpaths': did you mean one of the following?: Sub paths, Sub-paths, Subpar`

@Tim-Zhang I am trying again with `subPath` (between backticks). Hopefully that's sufficient.

**Comment by c3d** [comment]
It looks like the anchor generation does not insert `-` signs consistently, anchor is `kubernetes-volumemountssubpaths` and not `kubernetes-volumemounts-subpaths`. Fixed.

**Review by wainersm — APPROVED** [review]
LGTM. Thanks @c3d !

**Comment by fidencio** [comment]
Yeah, static check tests are not happy:
```
WARNING: Word 'subPaths': did you mean one of the following?: GOPATHs
WARNING: Incorrect word: 'volumeMounts'
ERROR: Spell check failed for file: 'docs/Limitations.md'
INFO: spell check failed for document docs/Limitations.md
ERROR: spell check failed, See https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md#spelling for more information.
```

**Comment by c3d** [comment]
I was hoping that words in backticks would not be spell checked. And indeed, what I see now are warnings, but in the end, it still marks the spell check as failed:

```
INFO: Checking document code blocks
15404
INFO: processing file 'docs/Limitations.md'
15405
INFO: Checking all document URLs
15406
INFO: Spell checking file 'docs/Limitations.md'
15407
WARNING: Word 'subPaths': did you mean one of the following?: GOPATHs
15408
WARNING: Incorrect word: 'volumeMounts'
15409
INFO: spell check failed for document docs/Limitations.md
15410
ERROR: Spell check failed for file: 'docs/Limitations.md'
15411
ERROR: spell check failed, See https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md#spelling for more information.
15412
Error: Process completed with exit code 1.
```

What is weird is that it works locally (on my own system):

```
% ../tests/cmd/check-spelling/kata-spell-check.sh check docs/Limitations.md
INFO: Spell checking file 'docs/Limitations.md'
INFO: Spell check successful for file: 'docs/Limitations.md'
```

I also see code that is supposed to remove any code block. I would assume this includes backticks. Checking.

**Comment by c3d** [comment]
Installing an Ubuntu 20.04 right now to check if that's part of the problem.

**Comment by c3d** [comment]
So interesting that the test fails in Ubuntu 20.04 but not Fedora 33:

```
 ../tests/cmd/check-spelling/kata-spell-check.sh check docs/Limitations.md
INFO: Spell checking file 'docs/Limitations.md'
WARNING: Word 'subPaths': did you mean one of the following?: GOPATHs
WARNING: Incorrect word: 'volumeMounts'
ERROR: Spell check failed for file: 'docs/Limitations.md
```

Need to fix the testing tool, because anything in backquotes should never be spell-checked.

**Comment by c3d** [comment]
My bad. Wrong branch (too many windows open). It fails on Fedora 33 the same way.

**Comment by c3d** [comment]
Filed https://github.com/kata-containers/tests/issues/3448, looking into it.

**Comment by fidencio** [comment]
@c3d, thanks, this one deserves a backport to the stable-2.0 branch.
